File: Trie.js
class TrieNode {
	constructor() {
		this.children = {};
		this.isWord = false;
	}

	addWord(word) {
		let curr = this;

		for (let ch of word) {
			if (!curr.children[ch]) {
				curr.children[ch] = new TrieNode();
			}

			curr = curr.children[ch];
		}

		curr.isWord = true;
	}

	search(word) {
		let curr = this

		for (let ch of word) {
			if (!curr.children[ch]) return false

			curr = curr.children[ch]
		}

		return curr.isWord

	}
}

function main() {
	let root = new TrieNode();
	let words = ['apple'];

	for (let w of words) {
		root.addWord(w);
	}


	console.log(root.search('apple'))
	console.log(root.search('app'))
}

main();

File: Trie.py
class TrieNode:
    def __init__(self):
        self.children = {}
        self.isWord = False
    
    def addWord(self, word):
        curr = self
        
        
        for ch in word:
            if ch not in curr.children:
                curr.children[ch] = TrieNode()
            
            curr = curr.children[ch]
            
        curr.isWord = True
        
        
            
if __name__ == '__main__':
    root = TrieNode()
    words = ['app', 'ape', 'ace']
    
    for w in words:
        root.addWord(w)
        
    
    